File: Task/Globally-replace-text-in-several-files/C-sharp/globally-replace-text-in-several-files.cs
using System.Collections.Generic;
using System.IO;

class Program {
    static void Main() {
        var files = new List<string> {
            "test1.txt",
            "test2.txt"
        };
        foreach (string file in files) {
            File.WriteAllText(file, File.ReadAllText(file).Replace("Goodbye London!", "Hello New York!"));
        }
    }
}

File: Task/Globally-replace-text-in-several-files/Delphi/globally-replace-text-in-several-files.pas
program Globally_replace_text_in_several_files;

{$APPTYPE CONSOLE}

uses
  System.SysUtils,
  System.IoUtils;

procedure StringReplaceByFile(_old, _new: string; FileName: TFilename;
  ReplaceFlags: TReplaceFlags = []); overload
var
  Text: string;
begin
  if not FileExists(FileName) then
    exit;
  Text := TFile.ReadAllText(FileName);
  TFile.Delete(FileName);
  TFile.WriteAllText(StringReplace(Text, _old, _new, ReplaceFlags), FileName);
end;

procedure StringReplaceByFile(_old, _new: string; FileNames: TArray<TFileName>;
  ReplaceFlags: TReplaceFlags = []); overload;
begin
  for var fn in FileNames do
    StringReplaceByFile(_old, _new, fn);
end;

begin
  StringReplaceByFile('Goodbye London!', 'Hello New York!', ['a.txt', 'b.txt', 'c.txt']);
end.
